src/Raven.Server/Documents/Indexes/Persistence/Corax/CoraxQueryBuilder.cs

```diff
@@ -81,7 +82,7 @@ public sealed class Parameters
                 ? new Lazy<List<string>>(() => IndexSearcher.GetFields())
                 : null;
             HasBoost = index.HasBoostedFields || query.Metadata.HasBoost ||
-                       (index.Configuration.OrderByScoreAutomaticallyWhenBoostingIsInvolved && query.Metadata.OrderBy is [{OrderingType: OrderByFieldType.Score} _, ..]); // in case when we've implicit boosting we've build primitives with scoring enabled
+                       (index.Configuration.OrderByScoreAutomaticallyWhenBoostingIsInvolved && query.Metadata.OrderBy?.Count(x => x.OrderingType == OrderByFieldType.Score) > 0); // in case when we've implicit boosting we've build primitives with scoring enabled
```

Comment:
You are now using LINQ and computation here, where before it didn't.
Probably better to re-write it without LInq.